griptape/artifacts/generic_artifact.py

```diff
@@ -9,7 +9,13 @@
 
 @define
 class GenericArtifact(BaseArtifact):
+    """Serves as an escape hatch for artifacts that don't fit into any other category.
+
+    Attributes:
+        value: The value of the Artifact.
+    """
+
     value: Any = field(metadata={"serializable": True})
 
```

Comment:
is this a good opportunity to update this artifact to `Artifact` and making it the top-level base? [this pattern](https://github.com/griptape-ai/griptape/blob/dev/griptape/configs/drivers/drivers_config.py) is [used](https://github.com/griptape-ai/griptape/blob/dev/griptape/memory/structure/conversation_memory.py) in a few other places.

Reply:
I think we should avoid subclassing concrete classes (even though we do it in this PR). So the other places are in violation of this and should be fixed IMO.

Reply:
any particular reason for that? Is subclassing concrete types a bad thing?

Reply:
It adds depth to the class hierarchy which can be difficult to maintain. There are times when it's unavoidable, but while we're still changing things around I think we should strive for a relatively flat hierarchy.